### PR TITLE
Track Slack threads only after successful assistant delivery

### PR DESCRIPTION
**Pull request by pragunseth23**
## Summary
- stop Slack app mentions from activating channel threads before the assistant successfully replies
- activate Slack threads from the assistant delivery path after a successful send or update, including resumed deliveries
- cover the delayed follow-up regression with gateway and assistant tests

Part of JARVIS-1046

### CODE REVIEW

**Comment by pragunseth23** [comment]
Review focus: this changes when Slack channel threads become active. The important edge cases are that bare app mentions no longer arm a thread, successful Slack sends and updates still do, and resumed deliveries stay able to reactivate the thread without creating ghost follow-ups. Risk is medium because it touches cross-package Slack ingress and delivery behavior, but the runtime and gateway regression tests should pin the intended flow.

File: assistant/src/runtime/channel-reply-delivery.ts [hunk 1]
```diff
@@ -89,6 +96,38 @@ function hasDeliverableReply(
...
+
+async function activateSlackThreadIfNeeded(
+  target: SlackActiveThreadTarget | null,
+): Promise<SlackActiveThreadTarget | null> {
+  if (!target) {
+    return null;
+  }
+
+  const tracked = await trackSlackActiveThread(target.channelId, target.threadTs);
+  return tracked ? null : target;
```
[the remark this answers is not in the document]
Comment (pragunseth23) [reply assistant/src/runtime/channel-reply-delivery.ts:128]: Addressed in the latest push: activation failures now surface as retryable delivery errors so Slack thread activation can be retried without silently dropping unmentioned follow-ups.
